File: app/src/main/java/awais/instagrabber/repositories/requests/directmessages/ReactionBroadcastOptions.java
package awais.instagrabber.repositories.requests.directmessages;

import java.util.HashMap;
import java.util.Map;

import awais.instagrabber.models.enums.BroadcastItemType;

public class ReactionBroadcastOptions extends BroadcastOptions {
    private final String itemId;
    private final boolean delete;

    public ReactionBroadcastOptions(final String clientContext,
                                    final ThreadIdOrUserIds threadIdOrUserIds,
                                    final String itemId,
                                    final boolean delete) {
        super(clientContext, threadIdOrUserIds, BroadcastItemType.REACTION);
        this.itemId = itemId;
        this.delete = delete;
    }

    @Override
    public Map<String, String> getFormMap() {
        final Map<String, String> form = new HashMap<>();
        form.put("item_id", itemId);
        form.put("reaction_status", delete ? "deleted" : "created");
        form.put("reaction_type", "like");
        return form;
    }
}

File: app/src/main/java/awais/instagrabber/repositories/requests/directmessages/ReactionBroadcastOptions.kt
package awais.instagrabber.repositories.requests.directmessages

import awais.instagrabber.models.enums.BroadcastItemType

class ReactionBroadcastOptions(
    clientContext: String,
    threadIdsOrUserIds: ThreadIdsOrUserIds,
    val itemId: String,
    val emoji: String?,
    val delete: Boolean
) : BroadcastOptions(clientContext, threadIdsOrUserIds, BroadcastItemType.REACTION) {
    override val formMap: Map<String, String>
        get() = listOfNotNull(
            "item_id" to itemId,
            "reaction_status" to if (delete) "deleted" else "created",
            "reaction_type" to "like",
            if (!emoji.isNullOrBlank()) "emoji" to emoji else null,
        ).toMap()
}